README.md
[![code coverage](https://codecov.io/gh/dwenzel/t3calendar/branch/master/graph/badge.svg)](https://codecov.io/gh/dwenzel/t3calendar)
[![Build Status](https://travis-ci.org/dwenzel/t3calendar.svg?branch=master)](https://travis-ci.org/dwenzel/t3calendar)  
[![SensioLabsInsight](https://insight.sensiolabs.com/projects/6a0eb348-f78f-4cc5-86f6-8a997cc9773b/small.png)](https://insight.sensiolabs.com/projects/6a0eb348-f78f-4cc5-86f6-8a997cc9773b)

Calendar
========

Calendar displays arbitrary items in a calendar view.  
This is an extension for the TYPO3 CMS. 


## Table of Contents

* [Introduction](./Documentation/Introduction.md)
* Configuration
    * [Paths](./Documentation/Configuration/Paths.md)
    * [Calendar](./Documentation/Configuration/Calendar.md)
* [ViewHelpers](./Documentation/ViewHelpers/ViewHelpers.md)
    * Format
        * [Date](./Documentation/ViewHelpers/Format/DateViewHelper.md)
    * Widget
        * [Calendar](./Documentation/ViewHelpers/Widget/CalendarViewHelper.md)
* Users Manual
    * [Cache](./Documentation/UsersManual/Cache.md)
* Developers Manual
    * [Custom Items](./Documentation/DevelopersManual/CustomItems.md)
    * [Custom Calendars](./Documentation/DevelopersManual/CustomCalendars.md)


ChangeLog
2019-07-08         [RELEASE] Release of t3calendar 0.7.1 (Dirk Wenzel)
2019-07-08 111437b [BUGFIX] fixed error in unit test caused by invalid method (Dirk Wenzel)

2019-07-08         [RELEASE] Release of t3calendar 0.7.0 (Dirk Wenzel)
2019-06-14 8b6368a [BUGFIX] dependencies in ext_emconf.php corrected (Dirk Wenzel)
2019-06-06 8f892de [TASK] include JavaScript (Dirk Wenzel)
2018-10-05 65d5803 [TASK] require typo3/cms-core:^8.7 || ^9.5 (Dirk Wenzel)
2018-05-15 d2f3b4f [TASK] add attribute rel=nofollow to all widget navigation links (Dirk Wenzel)

2018-04-10         [RELEASE] Release of t3calendar 0.6.0 (Dirk Wenzel)
2018-04-02 ddcd387 [TASK] CI: test matrix updated. - use typo3/minimal package for build - ext_emconf.php updated (Dirk Wenzel)
2018-04-02 bc78462 [TASK] compatibility for TYPO3 9 (and PHP 7.2) - use TYPO3 core ArrayUtility class instead of obsolete Extbase class - prevent error caused by invalid usage of count() (Dirk Wenzel)
2018-04-02 033ba44 [TASK] use mock builder instead of parent method 'getMock' in unit tests. (Dirk Wenzel)
2018-04-02 5dc4ed6 [TASK] require typo3/cms-core instead of typo3/cms (Dirk Wenzel)
2018-04-02 8f8fea9 [TASK] require typo3/cms >=8.7.0,<=9.99.99 (Dirk Wenzel)

2017-08-28         [RELEASE] Release of t3calendar 0.5.0 (Dirk Wenzel)
2017-08-05 7380791 [TASK][BREAKING] CalendarDay refactored. All occurrences of `calendarDay.isCurrent` in templates and partials where changed to `calendarDay.current` too. Make sure to adapt your custom templates! (Dirk Wenzel)
2017-08-05 4fdadff [TASK] coding guidelines (Dirk Wenzel)
2017-08-05 80db672 [TASK] CalendarDayFactory, CalendarDay, CalendarQuarter, CalendarYear, TemplateUtility refactored. (Dirk Wenzel)

2017-07-12         [RELEASE] Release of t3calendar 0.4.1 (Dirk Wenzel)
2017-07-12 62ff4f0 [BUGFIX] version number corrected. (Dirk Wenzel)
2017-06-23 3d2a385 [TASK] coding guidelines (Dirk Wenzel)

2017-05-12         [RELEASE] Release of t3calendar 0.4.1 (Dirk Wenzel)
2017-05-12 08abc8a [BUGFIX] fixed error which caused calendar items with end date being visible before their start date. (Dirk Wenzel)

2017-05-05         [RELEASE] Release of t3calendar 0.4.0 (Dirk Wenzel)
2017-05-05 9051f3d [BUGFIX] errors in unit test fixed (Dirk Wenzel)
2017-05-05 3c52151 [TASK] CalendarViewHelper now has an optional argument  ‘parameters’ of type array. Default templates adapted. (Dirk Wenzel)
2017-04-28 963aad0 [TASK] composer.json: obsolete settings removed. (Dirk Wenzel)
2017-04-28 759caab [BUGFIX] CalendarViewHelper: fixed issue caused by different implementation of AbstractWidgetViewHelper throughout TYPO3 versions. (Don't call parent::initializeArguments). (Dirk Wenzel)
2017-04-27 2a3ec3e [TASK] CalendarViewHelper refactored: register arguments explicitly. Allow passing an array as configuration. CalendarContoller now generates a configuration from array. Documentation added. (Dirk Wenzel)
2017-04-27 5586b06 [TASK] Documentation for path configuration added. (Dirk Wenzel)
2017-04-25 6d09007 [TASK] travis-ci: use custom UnitTests.xml instead of those from testing frame work (Dirk Wenzel)
2017-04-19 0dc58ce [TASK] fix travis-ci configuration (Dirk Wenzel)
2017-04-19 caa571a [TASK] require-dev nimut/testing-framework (Dirk Wenzel)

2017-03-27         [RELEASE] Release of t3calendar 0.3.0 (Dirk Wenzel)
2017-03-27 6f1740b [BUGFIX] missing css for loading indicator added. (Dirk Wenzel)
2017-03-27 3593d83 [BUGFIX] missing class for ajax navigation added to month template. (Dirk Wenzel)
2017-03-24 449d3a5 [TASK] Templates improved:  Ajax JavaScript is only active if Ajax is enabled for widget. Anchors added to navigation links. (Dirk Wenzel)
2017-03-11 0e79de6 [TASK] travis ci configuration: use unit test config from package. (Dirk Wenzel)

2017-03-11         [RELEASE] Release of t3calendar 0.2.0 (Dirk Wenzel)
2017-03-10 dd1eebf [TASK] CalendarItemStorage now respects valid end dates of items and adds them for each day in between date and end date. (Dirk Wenzel)
2017-03-10 629ffc6 [TASK] calendar items are now added to a calendar day if they have a matching end date. (Dirk Wenzel)
2017-03-10 87b2063 [TASK] Allow to enable caching of widget content in Extension Manager configuration. Caching is disabled by default  now. (Dirk Wenzel)
2017-03-09 cb32029 [TASK] TemplateUtility implemented. CalendarController refactored using it. (Dirk Wenzel)
2017-03-08 d34e794 [TASK] prefer source install of typo3/cms and typo3/cms-core (Dirk Wenzel)
2017-03-08 b673071 [TASK] travis ci configuration: adapt TYPO3  build directory by version (Dirk Wenzel)
2017-03-08 1506d7c [TASK] require phpunit/phpunit: >=4.8.0,<5.4.0 (Dirk Wenzel)
2017-03-08 b50bbd1 [TASK] obsolete class stub removed. (Dirk Wenzel)
2017-03-08 f2b8625 [TASK] register t3calendar content cache with TYPO3 pages group (Dirk Wenzel)
2017-01-24 54045f8 [TASK] CalendarController now initializes its view by setting template and partial root path (with an array of paths - thus allowing to override single template/partial files) (Dirk Wenzel)
2017-01-23 4536ce1 [TASK] pass settings to partial Calendar/Item (Dirk Wenzel)
2017-01-20 f46da01 [TASK] partial Calendar/Item added, wrapping of items in partial Month improved. (Dirk Wenzel)
2017-01-18 295dfe1 [TASK] Documentation: screen shots of views added. (Dirk Wenzel)
2017-01-18 ed113eb [BUGFIX] fixed error in month partial which caused display of wrong month when changing year. (Dirk Wenzel)
2017-01-18 f246bfd [TASK] Configuration added. Attributes for ajax functionality added to widget links in partial Month.html (Dirk Wenzel)
2017-01-18 5e7f3c7 [TASK] unit test for DateViewHelper added. (Dirk Wenzel)
2017-01-18 b2afaeb [TASK] unit test for DateViewHelper added. (Dirk Wenzel)
2017-01-18 3ed9ca3 [TASK] unit tests for DateViewHelper added. (Dirk Wenzel)
2017-01-18 d7e55fd [TASK] unit test for DateViewHelper added. (Dirk Wenzel)
2017-01-18 1e31961 [TASK] DateViewHelper ported from t3events. Templates adapted. (Dirk Wenzel)

ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "t3calendar".
 *
 * Auto generated 28-08-2017 09:14
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = array (
  'title' => 'Calendar',
  'description' => 'Calendar',
  'category' => 'plugin',
  'version' => '0.7.1',
  'state' => 'beta',
  'uploadfolder' => 0,
  'createDirs' => '',
  'clearcacheonload' => 1,
  'author' => 'Dirk Wenzel',
  'author_email' => 't3events@gmx.de',
  'constraints' =>
  array (
    'depends' =>
    array (
      'typo3' => '8.7.0-9.5.99'
    ),
    'conflicts' =>
    array (
    ),
    'suggests' =>
    array (
    ),
  ),
  '_md5_values_when_last_written' => 'a:101:{s:9:"ChangeLog";s:4:"88f0";s:9:"README.md";s:4:"189d";s:13:"composer.json";s:4:"66b4";s:21:"ext_conf_template.txt";s:4:"9b13";s:17:"ext_localconf.php";s:4:"399b";s:14:"ext_tables.php";s:4:"0ba8";s:35:"Classes/Cache/CacheManagerTrait.php";s:4:"166d";s:45:"Classes/Domain/Factory/CalendarDayFactory.php";s:4:"1d9d";s:54:"Classes/Domain/Factory/CalendarDayFactoryInterface.php";s:4:"a602";s:50:"Classes/Domain/Factory/CalendarDayFactoryTrait.php";s:4:"bca5";s:42:"Classes/Domain/Factory/CalendarFactory.php";s:4:"26ab";s:51:"Classes/Domain/Factory/CalendarFactoryInterface.php";s:4:"0909";s:47:"Classes/Domain/Factory/CalendarFactoryTrait.php";s:4:"212f";s:47:"Classes/Domain/Factory/CalendarMonthFactory.php";s:4:"febb";s:56:"Classes/Domain/Factory/CalendarMonthFactoryInterface.php";s:4:"9cdf";s:52:"Classes/Domain/Factory/CalendarMonthFactoryTrait.php";s:4:"0b14";s:49:"Classes/Domain/Factory/CalendarQuarterFactory.php";s:4:"9b0f";s:58:"Classes/Domain/Factory/CalendarQuarterFactoryInterface.php";s:4:"062c";s:54:"Classes/Domain/Factory/CalendarQuarterFactoryTrait.php";s:4:"287d";s:46:"Classes/Domain/Factory/CalendarWeekFactory.php";s:4:"165c";s:55:"Classes/Domain/Factory/CalendarWeekFactoryInterface.php";s:4:"c03a";s:51:"Classes/Domain/Factory/CalendarWeekFactoryTrait.php";s:4:"b820";s:46:"Classes/Domain/Factory/CalendarYearFactory.php";s:4:"519f";s:55:"Classes/Domain/Factory/CalendarYearFactoryInterface.php";s:4:"28af";s:51:"Classes/Domain/Factory/CalendarYearFactoryTrait.php";s:4:"1ea2";s:45:"Classes/Domain/Factory/ObjectManagerTrait.php";s:4:"1faa";s:33:"Classes/Domain/Model/Calendar.php";s:4:"b188";s:36:"Classes/Domain/Model/CalendarDay.php";s:4:"2b7c";s:46:"Classes/Domain/Model/CalendarItemInterface.php";s:4:"aaf2";s:38:"Classes/Domain/Model/CalendarMonth.php";s:4:"e208";s:40:"Classes/Domain/Model/CalendarQuarter.php";s:4:"281d";s:37:"Classes/Domain/Model/CalendarWeek.php";s:4:"8fdb";s:37:"Classes/Domain/Model/CalendarYear.php";s:4:"f978";s:35:"Classes/Domain/Model/MonthTrait.php";s:4:"e7bb";s:39:"Classes/Domain/Model/StartDateTrait.php";s:4:"f36c";s:50:"Classes/Domain/Model/Dto/CalendarConfiguration.php";s:4:"3cbc";s:57:"Classes/Domain/Model/Dto/CalendarConfigurationFactory.php";s:4:"2172";s:66:"Classes/Domain/Model/Dto/CalendarConfigurationFactoryInterface.php";s:4:"6849";s:62:"Classes/Domain/Model/Dto/CalendarConfigurationFactoryTrait.php";s:4:"594f";s:43:"Classes/Persistence/CalendarItemStorage.php";s:4:"7008";s:35:"Classes/Utility/TemplateUtility.php";s:4:"0504";s:45:"Classes/ViewHelpers/Format/DateViewHelper.php";s:4:"de9c";s:49:"Classes/ViewHelpers/Widget/CalendarViewHelper.php";s:4:"9f5e";s:60:"Classes/ViewHelpers/Widget/Controller/CalendarController.php";s:4:"6c47";s:38:"Configuration/TypoScript/constants.txt";s:4:"d41d";s:34:"Configuration/TypoScript/setup.txt";s:4:"5a91";s:29:"Documentation/Introduction.md";s:4:"7ce7";s:39:"Documentation/Configuration/Calendar.md";s:4:"a7f5";s:36:"Documentation/Configuration/Paths.md";s:4:"356f";s:55:"Documentation/DevelopersManual/CalendarConfiguration.md";s:4:"dd48";s:41:"Documentation/Images/comboViewQuarter.png";s:4:"2b79";s:38:"Documentation/Images/comboViewYear.png";s:4:"b9c5";s:34:"Documentation/Images/monthView.png";s:4:"5f60";s:34:"Documentation/UsersManual/Cache.md";s:4:"4e09";s:40:"Documentation/ViewHelpers/ViewHelpers.md";s:4:"c185";s:50:"Documentation/ViewHelpers/Format/DateViewHelper.md";s:4:"415f";s:54:"Documentation/ViewHelpers/Widget/CalendarViewHelper.md";s:4:"c844";s:44:"Resources/Private/Partials/Calendar/Day.html";s:4:"0b1e";s:45:"Resources/Private/Partials/Calendar/Item.html";s:4:"408d";s:46:"Resources/Private/Partials/Calendar/Month.html";s:4:"d4b0";s:53:"Resources/Private/Partials/Calendar/PeriodSwitch.html";s:4:"52be";s:48:"Resources/Private/Partials/Calendar/Quarter.html";s:4:"71f5";s:45:"Resources/Private/Partials/Calendar/Week.html";s:4:"d97c";s:45:"Resources/Private/Partials/Calendar/Year.html";s:4:"05d4";s:64:"Resources/Private/Templates/ViewHelpers/Widget/Calendar/Day.html";s:4:"f29f";s:66:"Resources/Private/Templates/ViewHelpers/Widget/Calendar/Index.html";s:4:"e4ab";s:66:"Resources/Private/Templates/ViewHelpers/Widget/Calendar/Month.html";s:4:"ede8";s:68:"Resources/Private/Templates/ViewHelpers/Widget/Calendar/Quarter.html";s:4:"66be";s:65:"Resources/Private/Templates/ViewHelpers/Widget/Calendar/Week.html";s:4:"82e0";s:65:"Resources/Private/Templates/ViewHelpers/Widget/Calendar/Year.html";s:4:"43d8";s:33:"Resources/Public/CSS/calendar.css";s:4:"fe40";s:39:"Resources/Public/JavaScript/calendar.js";s:4:"68fc";s:43:"Resources/Public/JavaScript/jquery-2.1.4.js";s:4:"d64c";s:25:"Tests/Build/UnitTests.xml";s:4:"2f36";s:42:"Tests/Unit/Cache/CacheManagerTraitTest.php";s:4:"22d8";s:52:"Tests/Unit/Domain/Factory/CalendarDayFactoryTest.php";s:4:"18c3";s:57:"Tests/Unit/Domain/Factory/CalendarDayFactoryTraitTest.php";s:4:"b289";s:49:"Tests/Unit/Domain/Factory/CalendarFactoryTest.php";s:4:"ec97";s:54:"Tests/Unit/Domain/Factory/CalendarFactoryTraitTest.php";s:4:"d54d";s:54:"Tests/Unit/Domain/Factory/CalendarMonthFactoryTest.php";s:4:"d339";s:59:"Tests/Unit/Domain/Factory/CalendarMonthFactoryTraitTest.php";s:4:"d1a5";s:56:"Tests/Unit/Domain/Factory/CalendarQuarterFactoryTest.php";s:4:"0503";s:61:"Tests/Unit/Domain/Factory/CalendarQuarterFactoryTraitTest.php";s:4:"210b";s:53:"Tests/Unit/Domain/Factory/CalendarWeekFactoryTest.php";s:4:"6825";s:58:"Tests/Unit/Domain/Factory/CalendarWeekFactoryTraitTest.php";s:4:"1bdc";s:53:"Tests/Unit/Domain/Factory/CalendarYearFactoryTest.php";s:4:"de24";s:58:"Tests/Unit/Domain/Factory/CalendarYearFactoryTraitTest.php";s:4:"5765";s:43:"Tests/Unit/Domain/Model/CalendarDayTest.php";s:4:"09a0";s:45:"Tests/Unit/Domain/Model/CalendarMonthTest.php";s:4:"8a9a";s:47:"Tests/Unit/Domain/Model/CalendarQuarterTest.php";s:4:"7615";s:40:"Tests/Unit/Domain/Model/CalendarTest.php";s:4:"6b23";s:44:"Tests/Unit/Domain/Model/CalendarWeekTest.php";s:4:"84fa";s:44:"Tests/Unit/Domain/Model/CalendarYearTest.php";s:4:"8c39";s:64:"Tests/Unit/Domain/Model/Dto/CalendarConfigurationFactoryTest.php";s:4:"a8b6";s:69:"Tests/Unit/Domain/Model/Dto/CalendarConfigurationFactoryTraitTest.php";s:4:"3c18";s:57:"Tests/Unit/Domain/Model/Dto/CalendarConfigurationTest.php";s:4:"151f";s:50:"Tests/Unit/Persistence/CalendarItemStorageTest.php";s:4:"7327";s:42:"Tests/Unit/Utility/TemplateUtilityTest.php";s:4:"8c90";s:52:"Tests/Unit/ViewHelpers/Format/DateViewHelperTest.php";s:4:"fc93";s:56:"Tests/Unit/ViewHelpers/Widget/CalendarViewHelperTest.php";s:4:"82a6";s:67:"Tests/Unit/ViewHelpers/Widget/Controller/CalendarControllerTest.php";s:4:"bc83";}',
);

